spec/controllers/review_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'spec_helper'
+
+describe ReviewsController do
+  describe 'POST create' do
+
+    let(:video) { Fabricate(:video) }
+    
+    context 'with authentication' do
+    
+      let(:user) { Fabricate(:user) }
+      
+      before do
+        session[:user_id] = user.id
+      end
+
+      context 'with valid input' do
+
+        before do
+          post :create, review: Fabricate.attributes_for(:review), video_id: video.id
+        end
+        
+        it 'should sets @video' do
+          expect(Video.first).to eq(video)
```

Comment:
This spec is flawed. You are fabricating this video on line 6, that means that it already exists in the db. This is not testing your create action. The proper way to test that would be to manually set a title in the fabricator on line 19 and then test for the title on the first video. By the way, this is an example of why you have to be careful with using let and before. The fabrication on line 6 was so far away that you didn't realize that it was affecting your spec. It is better to have all of the code your spec relies on in your spec, even if that means duplication.